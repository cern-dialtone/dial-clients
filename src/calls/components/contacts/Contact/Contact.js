import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Item, Icon } from 'semantic-ui-react';

import { formatUserOrganization } from 'calls/utils/formatters';
import styles from './Contact.module.css';

class Contact extends Component {
  static propTypes = {
    contact: PropTypes.shape({
      displayName: PropTypes.string.isRequired
    }).isRequired,
    selectContact: PropTypes.func.isRequired
  };

  selectContactAction = () => {
    const { contact, selectContact, findUserById } = this.props;
    selectContact(contact);
    findUserById(contact.personId);
  };

  render() {
    const { contact } = this.props;
    return (
      <Item className="" onClick={this.selectContactAction}>
        <div className={`ui tiny image ${styles.avatar}`}>
          <Icon
            name="user circle"
            size="big"
            color="blue"
            className="ui avatar"
          />
        </div>
        <Item.Content>
          <Item.Header className="">
            {contact ? contact.displayName : ''}
          </Item.Header>
          <Item.Extra>
            {contact ? formatUserOrganization(contact) : ''}
          </Item.Extra>
        </Item.Content>
      </Item>
    );
  }
}

export default Contact;
