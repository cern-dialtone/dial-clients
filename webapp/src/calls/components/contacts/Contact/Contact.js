import React, { Component } from "react";
import PropTypes from "prop-types";
import { Item, Icon } from "semantic-ui-react";

import styles from "./Contact.module.css";
import { formatUserOrganization } from "calls/utils/formatters";
import { logMessage } from "common/utils/logs";

class Contact extends Component {
  static propTypes = {
    contact: PropTypes.object.isRequired,
    selectContact: PropTypes.func.isRequired
  };

  selectContactAction = () => {
    const { contact, selectContact } = this.props;
    selectContact(contact);
  };

  render() {
    const { contact } = this.props;
    logMessage(contact);
    return (
      <Item className={``} onClick={this.selectContactAction}>
        <div className={`ui tiny image ${styles.avatar}`}>
          <Icon
            name="user circle"
            size={"big"}
            color={"blue"}
            className={"ui avatar"}
          />
        </div>
        <Item.Content>
          <Item.Header className={``}>
            {contact ? contact.displayName : ""}
          </Item.Header>
          <Item.Extra>
            {contact ? formatUserOrganization(contact) : ""}
          </Item.Extra>
        </Item.Content>
      </Item>
    );
  }
}

export default Contact;
