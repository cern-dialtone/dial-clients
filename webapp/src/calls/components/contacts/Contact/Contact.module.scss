.avatar{
  width: 20px !important;
}