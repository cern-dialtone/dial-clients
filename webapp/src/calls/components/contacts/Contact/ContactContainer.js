import { connect } from "react-redux";
import { selectContact } from "calls/actions/contacts";
import { bindActionCreators } from "redux";

import Contact from "calls/components/contacts/Contact/Contact";


function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      selectContact
    },
    dispatch
  );
}

export default connect(
    null,
    mapDispatchToProps
  )(Contact);
