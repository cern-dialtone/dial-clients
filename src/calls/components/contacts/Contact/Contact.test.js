import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import Contact from './Contact';

global.remote = {
  require: jest.fn()
};

describe('Contact tests', () => {
  it('renders empty List without any errors', async () => {
    const { asFragment } = render(
      <Contact
        selectContact={() => {}}
        contact={{
          displayName: 'Username'
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
