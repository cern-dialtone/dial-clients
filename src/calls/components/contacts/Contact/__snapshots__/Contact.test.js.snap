// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact tests renders empty List without any errors 1`] = `
<DocumentFragment>
  <div
    class="item"
  >
    <div
      class="ui tiny image avatar"
    >
      <i
        aria-hidden="true"
        class="blue user circle big icon ui avatar"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="header"
      >
        Username
      </div>
      <div
        class="extra"
      />
    </div>
  </div>
</DocumentFragment>
`;
